src/BSON/Document.c

```diff
@@ -196,23 +209,30 @@ static PHP_METHOD(MongoDB_BSON_Document, getIterator)
 	phongo_iterator_init(return_value, getThis());
 }
 
+static bool php_phongo_document_has(php_phongo_document_t* intern, char* key, size_t key_len)
+{
+	bson_iter_t iter;
+
+	if (!bson_iter_init(&iter, intern->bson)) {
+		phongo_throw_exception(PHONGO_ERROR_RUNTIME, "Could not initialize BSON iterator.");
```

Comment:
```suggestion
		phongo_throw_exception(PHONGO_ERROR_RUNTIME, "Could not initialize BSON iterator.");
		return false;
```

Otherwise you're accessing uninitialized memory with the subsequent `bson_iter_find_w_len()` call. Interesting that this didn't fail horribly in some test, but I suppose you may have lucked out and hit a guard check in [`_bson_iter_next_internal()`](https://github.com/mongodb/mongo-c-driver/blob/1.24.4/src/libbson/src/bson/bson-iter.c#L529).

Reply:
Ah, yes. Thanks for pointing this out.